internal/integration/api/apply-config.go

```diff
@@ -0,0 +1,169 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// +build integration_api
+
+package api
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"sort"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/talos-systems/talos/internal/integration/base"
+	machineapi "github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+	"github.com/talos-systems/talos/pkg/machinery/config"
+	"github.com/talos-systems/talos/pkg/machinery/config/configloader"
+	"github.com/talos-systems/talos/pkg/machinery/config/types/v1alpha1/machine"
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+// Sysctl to use for testing config changes.
+// This sysctl should:
+//
+//   - default to a non-Go-null value in the kernel
+//
+//   - not be defined by the default Talos configuration
+//
+//   - be generally harmless
+//
+const applyConfigTestSysctl = "net.ipv6.conf.all.accept_ra_mtu"
+
+const applyConfigTestSysctlVal = "1"
+
+const assertRebootedRebootTimeout = 10 * time.Minute
+
+type ApplyConfigSuite struct {
+	base.K8sSuite
+
+	ctx       context.Context
+	ctxCancel context.CancelFunc
+}
+
+// SuiteName ...
+func (suite *ApplyConfigSuite) SuiteName() string {
+	return "api.ApplyConfigSuite"
+}
+
+// SetupTest ...
+func (suite *ApplyConfigSuite) SetupTest() {
+	if testing.Short() {
+		suite.T().Skip("skipping in short mode")
+	}
+
+	// make sure we abort at some point in time, but give enough room for Recovers
+	suite.ctx, suite.ctxCancel = context.WithTimeout(context.Background(), 30*time.Minute)
+}
+
+// TearDownTest ...
+func (suite *ApplyConfigSuite) TearDownTest() {
+	if suite.ctxCancel != nil {
+		suite.ctxCancel()
+	}
+}
+
+// TestRecoverControlPlane removes the control plane components and attempts to recover them with the recover API.
+func (suite *ApplyConfigSuite) TestRecoverControlPlane() {
+	if !suite.Capabilities().SupportsReboot {
+		suite.T().Skip("cluster doesn't support reboot")
+	}
+
+	nodes := suite.DiscoverNodes().NodesByType(machine.TypeJoin)
+	suite.Require().NotEmpty(nodes)
+
+	sort.Strings(nodes)
+
+	node := nodes[0]
+
+	nodeCtx := client.WithNodes(suite.ctx, node)
+
+	provider, err := suite.readConfigFromNode(nodeCtx)
+	suite.Assert().Nilf(err, "failed to read existing config from node %q: %w", node, err)
```

Comment:
nit: you might want to change `suite.Assert()` to `suite.Require()` when the test can't continue running after the failure 

e.g. if config failed to be read, the next line will probably panic on `nil`, so it doesn't make much sense to proceed with the test after the failure